src/io/buffered.cr

```diff
@@ -33,9 +33,16 @@ module IO::Buffered
   def read_byte : UInt8?
     check_open
 
-    fill_buffer if @in_buffer_rem.empty?
+    fill_buffer if !sync? && @in_buffer_rem.empty?
     if @in_buffer_rem.empty?
-      nil
+      return nil unless sync?
+
+      byte = uninitialized UInt8
+      if read(Slice.new(pointerof(byte), 1)) == 1
```

Comment:
`byte if read(Slice.new(pointerof(byte), 1)) == 1`?

Reply:
I prefer my more explicit version.

Reply:
No. An `else nil` branch is always useless. Please drop it.

Reply:
Same for the `return nil` above. `return` *always* returns `nil`. No need for noise.

Reply:
I thought we agreed to skip the pointless bikeshedding about style.

Reply:
Also note that I copied this directly from io.cr: https://github.com/crystal-lang/crystal/blob/2d93603d453f42865fde421ae3e6af99eeaba2ad/src/io.cr#L302

Reply:
Formatter ain't a rewritter.

I never accepted the introduction of useless nils, and will always ask for them to be removed. Copying noise isn't a reason to keep it.